cogs/study.py

```diff
@@ -14,43 +26,295 @@ async def pcalc(self, ctx):
+        """Get the Answer to the Query."""
+        if query == "":
+            await ctx.send(f"Please submit a valid query! Like, `{constants.COMMAND_PREFIX}ask differentiate x+y^2 with respect to x`.")
+            return
+
+        temp_msg = await ctx.reply("Sending query, please wait.")
+
+        appid = s.WOLFRAM_APPID
+
+        try:
+            result = await self.get_query(query, appid)
+        except Exception:
+            await ctx.send("An unknown exception occurred while fetching query.")
+            return
+        if not result:
+            await ctx.soft_delete(temp_msg)
+            await ctx.send("Failed to get a response.")
+            return
+        if "queryresult" not in result:
+            await ctx.soft_delete(temp_msg)
+            await ctx.send("Did not get a valid response.")
+            return
+
+        link = f"[Display results online and refine query]({self.build_web_url(ctx.arg_str)})"
+        if not result["queryresult"]["success"] or result["queryresult"]["numpods"] == 0:
+            desc = f"Wolfram Alpha doesn't understand your query!\n Perhaps try rephrasing your question?\n{link}"
+            embed = discord.Embed(description=desc)
+            embed.set_footer(icon_url=ctx.author.avatar_url, text="Requested by {ctx.author}")
+            await ctx.soft_delete(temp_msg)
+            await ctx.offer_delete(await ctx.reply(embed=embed))
+            return
+
+        if ctx.flags["text"]:
+            fields = await self.pods_to_textdata(result["queryresult"]["pods"])
+            embed = discord.Embed(description=link)
+            embed.set_footer(icon_url=ctx.author.avatar_url, text=f"Requested by {ctx.author}")
+            await ctx.emb_add_fields(embed, fields)
+            await ctx.soft_delete(temp_msg)
+            out_msg = await ctx.send(embed=embed)
+            await ctx.offer_delete(out_msg)
+            return
+
+        important, extra = self.triage_pods(result["queryresult"]["pods"])
+
+        data = (await self.pods_to_filedata(important))[0]
+        output_data = [data]
 
-    @commands.command(aliases=['wq', 'wolframquestion', 'wquestion'])
-    async def ask_question(self, ctx, conversation_mode="true", *, question: str):
-        """Ask the answer of an question"""
+        embed = discord.Embed(description=link)
+        embed.set_footer(icon_url=ctx.author.avatar_url, text=f"Requested by {ctx.author}")
 
-        data = get_wolfram_data(question, conversation_mode)
+        await ctx.safe_delete_msgs([temp_msg])
+        out_msg = await ctx.send(file_data=data, file_name="wolf.png", embed=embed)
+        asyncio.ensure_future(ctx.offer_delete(out_msg))
 
-        embed = discord.Embed(title="Question Results")
+        if extra:
+            try:
+                await ctx.bot.add_reaction(out_msg, ctx.bot.objects["emoji_more"])
+            except discord.Forbidden:
+                pass
+            except discord.HTTPException:
+                pass
```

Comment:
Use `contextlib.suppress` instead:
```py
from contextlib import suppress


with suppress(discord.Forbidden, discord.HTTPException):
    await ctx.bot.add_reaction(out_msg, ctx.bot.objects["emoji_more"])
```
This is generally a preferred way to do it as `try-except` block should only be used when you want to do something in case exception occurs, when you're only using it with `pass`, `suppress` is a better alternative.